### DISCUSSION TITLE
Feedback: Describing GraphQL API

### DISCUSSION THREAD

**Discussion by nurbolnygmanov** [mswjs/mswjs.io#365] [Q&A]
- [Open page](https://mswjs.io/docs/network-behavior/graphql)

## Feedback

MSW does not work with Relay on testing side. Due to Relays own mockEnvironment wrapper.
so MSW can not intercept requests when running on server.

**Comment by kettanaito** [comment]
Hi, @nurbolnygmanov. Is the `mockEnvironment` mandatory? I assume it's a testing API, similar to `MockProvider` in Apollo. If that's the case, you _mustn't_ be using it when adding MSW to the mix. MSW is the substitute for all those mock environments and providers. 

I don't have much context on what and how Relay does, unfortunately. You can bring it up with the Relay team on GtiHub or other platforms and ask around what exactly `mockEnvironment` is doing to the environment. If MSW doesn't work with it, it means `mockEnvironment` is patching request-issuing modules and prevent the requests from actually happening. We discourage that in MSW and no solutions that do that will work with it.
